assets/js/util/index.js

```diff
@@ -135,23 +135,26 @@ export const readableLargeNumber = ( number, currencyCode = false ) => {
 	};
 
 	switch ( true ) {
-		case 1000000 < number :
+		// Numbers over 1,000,000 round normally and display a single decimal unless the decimal is 0.
+		case 1000000 <= number :
 			return sprintf(
 				// translators: %s: an abbreviated number in millions.
 				__( '%sM', 'google-site-kit' ),
-				numberFormat( number / 1000000, withSingleDecimal )
+				numberFormat( number / 1000000, number % 10 === 0 ? {} : withSingleDecimal )
 			);
-		case 99000 < number :
+		// Numbers between 10,000 and 1,000,000 round normally and have no decimals
+		case 10000 <= number :
 			return sprintf(
 				// translators: %s: an abbreviated number in thousands.
 				__( '%sK', 'google-site-kit' ),
 				numberFormat( Math.round( number / 1000 ) )
 			);
-		case 1000 < number :
+		// Numbers between 1000 and 10,000 round normally and display a single decimal unless the decimal is 0.
```

Comment:
```suggestion
		// Numbers between 1,000 and 10,000 round normally and display a single decimal unless the decimal is 0.
```

Just to be consistent 😉